### PR TITLE
Add GitHub Pullrequest documentation

### PR DESCRIPTION
**Pull request by olblak**
Fix: https://github.com/updatecli/updatecli/issues/488

I didn't introduce the concept of action as discussed [here](https://github.com/updatecli/updatecli/issues/465) as it's not been implemented yet. 
 
Signed-off-by: Olblak <me@olblak.com>

### CODE REVIEW

File: content/docs/plugins/github_pullrequest.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+---
+
+// <!-- Required for asciidoctor -->
+:toc:
+// Set toclevels to be at least your hugo [markup.tableOfContents.endLevel] config key
+:toclevels: 4
+
+== Description
+
+The pullrequest resource describes how updatecli handle GitHub pullrequest.
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:25]:
```suggestion
The `pullrequests` section describes the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests[GitHub Pull Requests] that updatecli is expected to open (or update) when the manifest is applied.
```

- It's not a resource ;)
- "describes how" is a wrong assertion: the whole goal of describing is to abstract the "how" away from user (vs. scripting)
- Not everyone is at ease with pull request: adding a link to official GitHub doc

File: content/docs/plugins/github_pullrequest.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+== Parameters
+
+[cols="1,1,1,4",options=header]
+|===
+| Name | Required | Default |Description
+| automerge | |-| Enable automerge. More on link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub documentation]
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:32]:
```suggestion
| automerge | |-| Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the Pull Request
```

- Given that we discussed a lot about the expected meaning of wording "automerge", WDYT about making it explicit that it is the GitHub automerge feature?

File: content/docs/plugins/github_pullrequest.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+| description | |-| Insert pullrequest text before the one generated by updatecli
+| draft | |-| Disallow maintainer to modify the pullrequest
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:34]:
```suggestion
| draft | |-| Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[Draft state (as defined by GitHub)]
```

File: content/docs/plugins/github_pullrequest.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+The pullrequest resource describes how updatecli handle GitHub pullrequest.
+
+== Parameters
+
+[cols="1,1,1,4",options=header]
+|===
+| Name | Required | Default |Description
+| automerge | |-| Enable automerge. More on link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub documentation]
+| description | |-| Insert pullrequest text before the one generated by updatecli
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:33]: Can you elaborate on this one, I don't understand what it does?

File: content/docs/plugins/github_pullrequest.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+| draft | |-| Disallow maintainer to modify the pullrequest
+| labels | |-| Set PullRequest labels
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:35]:
```suggestion
| labels | |-| Set the labels of this pull request (list of labels)
```

File: content/docs/plugins/github_pullrequest.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+| maintainercannotmodify | |-| Disallow maintainer to modify the pullrequest
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:36]: Did not know about this one. Is there more details on GitHub documentation to help understanding?

File: content/docs/plugins/github_pullrequest.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+| title | |-| Set pullrequest title
```
Comment (dduportal) [inline content/docs/plugins/github_pullrequest.adoc:37]:
```suggestion
| title | |-| Set the title of the pull request
```

**Review by dduportal — CHANGES_REQUESTED** [review]
Cool thing, this change will help a lot!

A few feedbacks to help making it understandable by non updatecli maintainers :)

File: content/docs/plugins/github_pullrequest.adoc (hunk 7 above)
Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc:37]: I prefer the shorter version, is there a requirement for the longer one?

File: content/docs/plugins/github_pullrequest.adoc (hunk 6 above)
Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc:36]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

File: content/docs/plugins/github_pullrequest.adoc (hunk 7 above)
Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:37]: > I prefer the shorter version, is there a requirement for the longer one?

none except "pull request in 2 words

Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:37]:
```suggestion
| title | |-| Set the pull request title
```

WDYT?

File: content/docs/plugins/github_pullrequest.adoc (hunk 6 above)
Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:36]: oh thanks! then:

```suggestion
| maintainercannotmodify | |-| Forbid maintainers to link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests[edit the pull request]
```

File: content/docs/plugins/github_pullrequest.adoc (hunk 4 above)
Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:33]:
```suggestion
| description | | "" | Insert pullrequest text before the one generated by updatecli
```

File: content/docs/plugins/github_pullrequest.adoc (hunk 2 above)
Comment (olblak) [reply content/docs/plugins/github_pullrequest.adoc:32]: Thanks for the suggestio, I like it very much. Also thanks for the default value which I forgot to put

File: content/docs/plugins/github_pullrequest.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+lastmod: 2022-01-30T08:21:01+02:00
+draft: false
+images: []
+menu: 
+  docs:
+    parent: "plugins"
+weight: 130 
+toc: true
+plugins:
+  - pullrequest
```
Comment (olblak) [inline content/docs/plugins/github_pullrequest.adoc:15]: @dduportal @lemeurherve For the time being, I am considering this "resource" a plugin for  "pull request". Once the "action" concept is ready to be used then we would just change this line to "action".
The only intends is to highlight this documentation in the correct doc menu

File: content/docs/plugins/github_pullrequest.adoc (hunk 7 above)
Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc]: I think "pull request" in 2 words should be applied everywhere then :)

File: content/docs/plugins/github_pullrequest.adoc (hunk 8 above)
Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc:15]: LGTM

File: content/docs/plugins/github_pullrequest.adoc (hunk 7 above)
Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc]: Yes, i agree

Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc]: (Except on the yaml of course)

File: content/docs/plugins/github_pullrequest.adoc (hunk 8 above)
Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:15]: Lgtm too , we are in sync

File: content/docs/plugins/github_pullrequest.adoc [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+== Description
+
+The `pullrequests` section describes the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests[GitHub Pull Requests] that updatecli is expected to open (or update) when the manifest is applied.
+
+== Parameters
+
+[cols="1,1,1,4",options=header]
+|===
+| Name | Required | Default |Description
+| automerge | | false | Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the Pull Request
```
Comment (lemeurherve) [inline content/docs/plugins/github_pullrequest.adoc:32]:
```suggestion
| automerge | | false | Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the pull request
```

File: content/docs/plugins/github_pullrequest.adoc [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+| description | | "" | Insert pullrequest text before the one generated by updatecli
```
Comment (lemeurherve) [inline content/docs/plugins/github_pullrequest.adoc:33]:
```suggestion
| description | | "" | Insert the pull request text before the one generated by updatecli
```

File: content/docs/plugins/github_pullrequest.adoc [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+| draft | | false | Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[Draft state (as defined by GitHub)]
```
Comment (lemeurherve) [inline content/docs/plugins/github_pullrequest.adoc:34]:
```suggestion
| draft | | false | Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[draft state (as defined by GitHub)]
```

File: content/docs/plugins/github_pullrequest.adoc (hunk 10 above)
Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc:33]: I find this not really clear.

Comment (olblak) [reply content/docs/plugins/github_pullrequest.adoc:33]: What would you think of:
```suggestion
| description | | "" | Prepend a custom message to the pull request body generated by updatecli
```

Comment (lemeurherve) [reply content/docs/plugins/github_pullrequest.adoc:33]:
```suggestion
| description | | "" | Prepend a custom description to the pull request body message generated by updatecli
```

Comment (dduportal) [reply content/docs/plugins/github_pullrequest.adoc:33]: 👍  (both proposals)

Comment (olblak) [reply content/docs/plugins/github_pullrequest.adoc:33]: I prefer @lemeurherve 's one

**Review by dduportal — APPROVED** [review]

**Comment by dduportal** [comment]
Good job 👏

**Pull request merged**